auth-foundation/build.gradle.kts

```diff
@@ -0,0 +1,102 @@
+plugins {
+    id("com.android.library")
+    id("com.google.devtools.ksp")
+    id("kotlin-parcelize")
+    id("org.jetbrains.dokka")
+    id("spotless")
+    kotlin("android")
+    kotlin("plugin.serialization") version libs.versions.kotlin.get()
+    id("com.vanniktech.maven.publish.base")
+}
+
+android {
+    namespace = "com.okta.authfoundation"
+    compileSdk = COMPILE_SDK
+
+    defaultConfig {
+        minSdk = MIN_SDK
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles("consumer-rules.pro")
+
+        ksp {
+            arg("room.generateKotlin", "true")
+            arg("room.schemaLocation", "$projectDir/schemas")
+        }
+
+        buildConfigField("String", "SDK_VERSION", "\"okta-auth-foundation-kotlin/$AUTH_FOUNDATION_VERSION\"")
```

Comment:
In the other libraries, I've been trying to use the term "authfoundation" as a single word, and to have libraries be separated by dashes between "okta", the name of the library, and the platform.  So we'd have library names such as:

* AuthFoundation -- `okta-authfoundation-{swift,kotlin,javascript,dotnet}`
* DirectAuth -- `okta-directauth-{swift,kotlin,javascript,dotnet}`
* OAuth2 -- `okta-oauth2-{swift,kotlin,javascript,dotnet}`

Does that work within the Android/Java ecosystem?  We don't really need to do anything about this now (I know this would be a big change), but I wanted to bring it up.

Reply:
okta-authfoundation works with Android/Java ecosystem. And our package namespaces actually align more closely with that naming scheme anyways.

Reply:
@alexnachbaur-okta We can discuss and update in another PR. Changing this will have impact the current useragent. I will have to go over the impact it has in any of our monitoring. This PR is for updating the gradle build scripts.